lib/linter/apply-disable-directives.js

```diff
@@ -199,6 +199,88 @@ function processUnusedDisableDirectives(allDirectives) {
     );
 }
 
+/**
+ * Collect eslint-enable comments that are used after removing the suppression by eslint-disable comments.
+ * @param {Directive[]} directives The directives to check.
+ * @returns {Iterable<Directive>} The used eslint-enable comments
+ */
+function *collectUsedEnableDirectives(directives) {
+    const withoutRuleIdKey = Symbol("withoutRuleId");
+    const enabledRules = new Map();
+
+    for (let index = directives.length - 1; index >= 0; index--) {
+        const directive = directives[index];
+
+        if (directive.type === "disable") {
+            if (enabledRules.size === 0) {
+                continue;
+            }
+            if (directive.ruleId === null) {
+
+                // If encounter `eslint-disable` without ruleId,
+                // mark all `eslint-enable` currently held in enabledRules as used.
+                // e.g.
+                //    /* eslint-disable */ <- current directive
+                //    /* eslint-enable rule-id1 */ <- used
+                //    /* eslint-enable rule-id2 */ <- used
+                //    /* eslint-enable */ <- used
+                for (const enableDirective of enabledRules.values()) {
+                    yield enableDirective;
+                }
```

Comment:
```suggestion
                yield* enabledRules.values();
```

I think it can be simplified this way.

Though, this generator as currently implemented can yield duplicates (e.g., for `/* eslint-disable no-undef, no-alert */ /* eslint-enable */` it yields the same enable directive twice), so maybe it would make sense to collect directives in a Set and convert this generator to a function that returns that Set?

Reply:
@mdjermanovic Thank you. I applied the suggestion.
I considered using a Set to make it unique, but I don't think it makes much sense since the side that uses that iterator is already using a Set. However, I do think it's easier to read if it's already unique in the function. Do you think it is better to make the `yield` value unique?

Reply:
I think it would be clearer if this doesn't yield/return duplicates. When I noticed that it yields duplicates in some cases, I had to double-check whether duplicates can affect the code that uses it in an unexpected way, so it seems better to avoid thinking about that and possible problems in the future if we refactor the code in a way that doesn't expect duplicates.

If we'd use a Set to make it unique, then this could be a function that returns that Set?